pkg/webhook/http.go

```diff
@@ -0,0 +1,125 @@
+package webhook
+
+import (
+	"encoding/json"
+	blackduckclientset "github.com/blackducksoftware/synopsys-operator/pkg/blackduck/client/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"net/http"
+	"time"
+
+	"k8s.io/api/admission/v1beta1"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+// OperatorWebhook is used to create the admission webhook
+type OperatorWebhook struct {
+	kubeConfig      *rest.Config
+	kubeClient      *kubernetes.Clientset
+	blackduckClient *blackduckclientset.Clientset
+}
+
+// NewOperatorWebhook will return an OperatorWebhook
+func NewOperatorWebhook(kubeConfig *rest.Config) *OperatorWebhook {
+	kubeclient, err := kubernetes.NewForConfig(kubeConfig)
```

Comment:
we have method to provide the kubeclient and kubeconfig in https://github.com/blackducksoftware/synopsys-operator/blob/master/pkg/protoform/kubeconfig.go

Reply:
Correct, but calling those functions multiple times when using the out of cluster configuration will cause issues because of the Flags.  This is why we re-use the kubeconfig

Reply:
I think @mphammer is fixing the issue related to flags.. Whether it will fix the issue over here as well?